src/broker/broker.c

```diff
@@ -847,6 +847,43 @@ receiver_thr_f (void *arg)
 	  continue;
 	}
 
+      if (strncmp (cas_req_header, "STATUS", 6) == 0)
+	{
+	  int status = -1;
+	  int pid, i, len;
+	  unsigned int session_id;
+
+	  len = read_nbytes_from_client (clt_sock_fd, cas_req_header + SRV_CON_CLIENT_INFO_SIZE, 4);
+	  if (len < 0)
+	    {
+	      CLOSE_SOCKET (clt_sock_fd);
+	      continue;
+	    }
+
+	  memcpy ((char *) &pid, cas_req_header + 6, 4);
+	  pid = ntohl (pid);
+	  memcpy ((char *) &session_id, cas_req_header + 10, 4);
+	  session_id = ntohl (session_id);
+
+	  if (shm_br->br_info[br_index].shard_flag == OFF)
+	    {
+	      for (i = 0; i < shm_br->br_info[br_index].appl_server_max_num; i++)
+		{
+		  if (shm_appl->as_info[i].service_flag == SERVICE_ON && shm_appl->as_info[i].pid == pid)
+		    {
+		      if (session_id == shm_appl->as_info[i].session_id)
+			{
+			  status = shm_appl->as_info[i].fn_status;
+			}
+		      break;
+		    }
+		}
+	      CAS_SEND_ERROR_CODE (clt_sock_fd, status);
+	      CLOSE_SOCKET (clt_sock_fd);
+	      continue;
+	    }
```

Comment:
if shard_flag <> OFF then sock_fd is not closed.

Reply:
shard_flag is used only version 9, so the shard_flag is always off.
And the socket is closed in JDBC after receiving the result from the broker.